zingolib/src/wallet/transaction_records_by_id/trait_inputsource.rs

```diff
@@ -38,6 +38,68 @@ pub enum InputSourceError {
     InvalidValue(BalanceError),
 }
 
+// Calculate remaining difference between target and selected.
+// There are two mutually exclusive cases:
+//    (Change) There's no more needed so we've selected 0 or more change
+//    (Positive) We need > 0 more value.
+// This function represents the NonPositive case as None, which
+// then serves to signal a break in the note selection for where
+// this helper is uniquely called.
+fn calculate_remaining_needed(
```

Comment:
i dont understand the purpose of this function and enum when the end result actually is more complicated and has more lines of code in the main loop for no gain AFAIK